### PR TITLE
Weather api

### PR DESCRIPTION
**Pull request by MisaOgura**

### CODE REVIEW

File: app/controllers/weatherapis_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require_relative '../../lib/weather_api'
+
+class WeatherapisController < ApplicationController
```
Comment (shaneoston72) [inline app/controllers/weatherapis_controller.rb]: change this to WeatherApiController?

File: app/helpers/weatherapis_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module WeatherapisHelper
+end
```
Comment (rhiannonruth) [inline app/helpers/weatherapis_helper.rb]: what does this file do?

File: app/helpers/weatherapis_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module WeatherapisHelper
```
Comment (shaneoston72) [inline app/helpers/weatherapis_helper.rb]: chang module name to WeatherApiHelper

File: lib/weather_api.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  include HTTParty
+
+  def initialize
+    @base_uri = 'http://api.openweathermap.org/data/2.5/weather'
+    @options = { city_name: 'London', api_key: ENV['WEATHER_API_KEY'] }
+    @raw_json = ''
+  end
+
+  def grab_weather_json
+    @raw_json = self.class.get( @base_uri +
```
Comment (rhiannonruth) [inline lib/weather_api.rb]: could you remove the space before the @ sign please?

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
 require 'helper_methods'
+require 'webmock/rspec'
+require 'json'
+
+WebMock.disable_net_connect!(allow_localhost: true)
 
 RSpec.configure do |config|
 
+  config.before(:each) do
+    mock_body =  { "weather"=>[{ "id":801, "description"=>"few clouds" }],
```
Comment (rhiannonruth) [inline spec/spec_helper.rb]: can we be consistent with using : or => in the hash please?

**Comment by shaneoston72** [comment]
see comments please...

File: lib/weather_api.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                               "?q=#{@options[:city_name]}" +
+                               "&appid=#{@options[:api_key]}").parsed_response
+  end
+
+  def show_desc_and_temp
+    desc = @raw_json['weather'][0]['description']
+    temp = (@raw_json['main']['temp'].to_i - 273.15).round(1)
+    {description: desc, temperature: temp}.to_json
```
Comment (rhiannonruth) [inline lib/weather_api.rb]: can we have spaces after/before the {/} please?

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
+                   "main"=>{"temp"=>291.563}
+                 }.to_json
+    weather_api_key = ENV['WEATHER_API_KEY']
+    stub_request(:get, "http://api.openweathermap.org/data/2.5/weather?q=London&appid=" + weather_api_key).
```
Comment (rhiannonruth) [inline spec/spec_helper.rb]: I think it's a little clearer to have the . on the line below, like so:

```
stub_request(:get, "http://api.openweathermap.org/data/2.5/weather?q=London&appid=" + weather_api_key)
  .with(:headers => {'Accept'=>'*/*', 
                     'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
                     'User-Agent'=>'Ruby'})
  .to_return(:status  => 200, 
             :body    => mock_body, 
             :headers => {})
```

File: spec/spec_helper.rb (hunk 5 above)
Comment (MisaOgura) [inline spec/spec_helper.rb]: this is the format that comes back from the real weather API so I thought I'd keep their format!

File: app/helpers/weatherapis_helper.rb (hunk 2 above)
Comment (MisaOgura) [inline app/helpers/weatherapis_helper.rb]: automatically created with `rails g controller` command, can remove!

File: lib/weather_api.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+require 'dotenv'
+require 'json'
+Dotenv.load
+
+class WeatherApiCall
+
+  include HTTParty
+
+  def initialize
+    @base_uri = 'http://api.openweathermap.org/data/2.5/weather'
```
Comment (rhiannonruth) [inline lib/weather_api.rb]: is this meant to be url not uri?

**Comment by rhiannonruth** [comment]
:shipit:

**Comment by shaneoston72** [comment]
:shipit:

**Pull request merged**